### PR TITLE
Add QRegion support

### PR DESCRIPTION
**Pull request by Montel**

### CODE REVIEW

File: crates/cxx-qt-lib-headers/src/lib.rs [hunk 1]
```diff
@@ -50,6 +50,8 @@ pub fn build_opts() -> cxx_qt_build::CxxQtBuildersOpts {
         (include_str!("../include/core/qpoint.h"), "qpoint.h"),
         (include_str!("../include/core/qpointf.h"), "qpointf.h"),
         (include_str!("../include/core/qrect.h"), "qrect.h"),
+        #[cfg(feature = "qt_gui")]
+        (include_str!("../include/gui/qregion.h"), "qregion.h"),
```
Comment (LeonMatthesKDAB) [inline crates/cxx-qt-lib-headers/src/lib.rs:54]: Please move this down to the other `#[cfg(feature="qt_qui")]` lines.

File: crates/cxx-qt-lib/src/gui/qregion.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+        /// Returns a region which is r subtracted from this region.
+        fn subtracted(self: &QRegion, r: &QRegion) -> QRegion;
+
+        /// Translates the region point.x() along the x axis and point.y() along the y axis, relative to the current position.
+        /// Positive values move the region to the right and down.
+        /// Translates to the given point.
+        fn translate(self: &mut QRegion, point: &QPoint);
+
+        /// Returns a copy of the regtion that is translated p.x() along the x axis and p.y() along the y axis,
```
Comment (LeonMatthesKDAB) [inline crates/cxx-qt-lib/src/gui/qregion.rs:51]:
```suggestion
        /// Returns a copy of the region that is translated p.x() along the x axis and p.y() along the y axis,
```

**Review by LeonMatthesKDAB — COMMENTED** [review]
Overall looks good, some minor nitpicks :)

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**